File: python/leetcode20.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack=[]
        ch=''
        for i in range(len(s)):
            if s[i]=="(" or s[i]=="[" or s[i]=="{":
                stack.append(s[i])
            else:
                if not stack:
                    return False
                if stack[-1]=="{" and s[i]=="}" or\
                   stack[-1]=="(" and s[i]==")" or\
                   stack[-1]=="[" and s[i]=="]":
                       stack.pop()
                       continue
                else:
                    return False
            
        if stack:
            return False
        else:
            return True
    

File: java/leetcode20.java
/*
 * @lc app=leetcode id=20 lang=java
 *
 * [20] Valid Parentheses
 */

// @lc code=start
import java.util.*;
class Solution {
    public boolean isValid(String s) {
        Stack<Character> stack = new Stack<>();

        for(int i=0;i<s.length();i++){
            if (s.charAt(i)=='(' || s.charAt(i)=='{' || s.charAt(i)=='['){
                stack.push(s.charAt(i));
            }else{
                if(stack.size()==0){
                    return false;
                }
                if((s.charAt(i)==')' && stack.peek()=='(') || 
                (s.charAt(i)=='}' && stack.peek()=='{') ||
                (s.charAt(i)==']' && stack.peek()=='[')){
                    stack.pop();
                    continue;
                }else{
                    return false;
                }
            }
        }
        if (stack.isEmpty()){

        return true;
        }else{
            return false;
        }
    }
}
// @lc code=end
